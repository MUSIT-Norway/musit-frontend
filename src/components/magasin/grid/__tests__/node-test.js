import assert from 'assert';
import React from 'react';
import ReactTestUtils from 'react-addons-test-utils';
import NodeLeftMenuComponent from '../LeftMenu';

describe('NodeLeftMenuComponent', () => {
  let labels;
  let buttons;
  const setup = () => {
    const myDiv = ReactTestUtils.renderIntoDocument(
      <NodeLeftMenuComponent
        rootNode={{ id: 1}}
        translate={(key) => key}
        onClickNewNode={(key) => key}
        objectsOnNode={11}
        totalObjectCount={78}
        underNodeCount={5}
        showButtons
        onClickProperties={(key) => key}
        onClickControlObservations={(key) => key}
        onClickObservations={(key) => key}
        onClickController={(key) => key}
        onClickMoveNode={(key) => key}
        onClickDelete={(key) => key}
      />
    );
    labels = ReactTestUtils.scryRenderedDOMComponentsWithTag(myDiv, 'label');
    buttons = ReactTestUtils.scryRenderedDOMComponentsWithTag(myDiv, 'button');
  };
  it('Check the New newNode button is created.', () => {
    setup();
    assert(buttons[0].getAttribute('id') === '1_newNode');
  });
  it('Check the New properties button is created.', () => {
    setup();
    assert(buttons[1].getAttribute('id') === '1_properties');
  });
  it('Check the objectsOnNode label is created.', () => {
    setup();
    assert(labels[0].getAttribute('id') === '1_objectsOnNode');
  });
  it('Check the totalObjectCount label is created.', () => {
    setup();
    assert(labels[1].getAttribute('id') === '1_totalObjectCount');
  });
});
