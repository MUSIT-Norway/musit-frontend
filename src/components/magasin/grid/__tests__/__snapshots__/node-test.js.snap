exports[`NodeLeftMenuComponent renders properly 1`] = `
<div>
  <div
    style={
      Object {
        "border": "none",
        "textAlign": "center"
      }
    }>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      id="1_newNode"
      onClick={[Function onClick]}
      style={
        Object {
          "textAlign": "left"
        }
      }>
      <FontAwesome
        name="plus-circle"
        style={
          Object {
            "padding": "2px"
          }
        } />
      Ny node
    </Button>
  </div>
  <hr />
  <div
    style={
      Object {
        "border": "none",
        "textAlign": "center"
      }
    }>
    Num Nodes
    <br />
    <ControlLabel
      bsClass="control-label"
      id="1_numNodes"
      srOnly={false}>
      11
    </ControlLabel>
  </div>
  <div
    style={
      Object {
        "border": "none",
        "textAlign": "center"
      }
    }>
    Num Objects
    <br />
    <ControlLabel
      bsClass="control-label"
      id="1_numObjects"
      srOnly={false}>
      5
    </ControlLabel>
  </div>
  <div
    style={
      Object {
        "border": "none",
        "textAlign": "center"
      }
    }>
    Total Objects
    <br />
    <ControlLabel
      bsClass="control-label"
      id="1_totalObjects"
      srOnly={false}>
      78
    </ControlLabel>
  </div>
  <hr />
  <div
    style={
      Object {
        "border": "none",
        "textAlign": "center"
      }
    }>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="link"
      disabled={false}
      id="1_properties"
      onClick={[Function onClick]}
      style={
        Object {
          "color": "black"
        }
      }>
      <FontAwesome
        name="cog"
        style={
          Object {
            "fontSize": "1.5em",
            "padding": "2px"
          }
        } />
      <br />
      Egenskaper
    </Button>
  </div>
  <div
    style={
      Object {
        "border": "none",
        "textAlign": "center"
      }
    }>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="link"
      disabled={false}
      id="1_controlsobservations"
      onClick={[Function onClick]}
      style={
        Object {
          "color": "black"
        }
      }>
      <span
        className="icon icon-musitcontrolobsicon"
        style={
          Object {
            "padding": "2px"
          }
        } />
      <br />
      Kontr./Observ.
    </Button>
  </div>
  <div
    style={
      Object {
        "border": "none",
        "textAlign": "center"
      }
    }>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="link"
      disabled={false}
      id="1_moveNode"
      onClick={[Function onClick]}
      style={
        Object {
          "color": "black"
        }
      }>
      <FontAwesome
        name="truck"
        style={
          Object {
            "fontSize": "1.5em",
            "padding": "2px"
          }
        } />
      <br />
      Flytt node
    </Button>
  </div>
  <div
    style={
      Object {
        "border": "none",
        "textAlign": "center"
      }
    }>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="link"
      disabled={true}
      id="1_delete"
      onClick={[Function onClick]}
      style={
        Object {
          "color": "black"
        }
      }>
      <FontAwesome
        name="trash-o"
        style={
          Object {
            "fontSize": "1.5em",
            "padding": "2px"
          }
        } />
      <br />
      Slett
    </Button>
  </div>
</div>
`;
